numpy/core/src/umath/loops.c.src

```diff
@@ -1642,26 +1642,26 @@ pairwise_sum_@TYPE@(@dtype@ *a, npy_uintp n, npy_intp stride)
          * 8 times unroll reduces blocksize to 16 and allows vectorization with
          * avx without changing summation ordering
          */
-        r[0] = @trf@(a[0 * stride]);
-        r[1] = @trf@(a[1 * stride]);
-        r[2] = @trf@(a[2 * stride]);
-        r[3] = @trf@(a[3 * stride]);
-        r[4] = @trf@(a[4 * stride]);
-        r[5] = @trf@(a[5 * stride]);
-        r[6] = @trf@(a[6 * stride]);
-        r[7] = @trf@(a[7 * stride]);
+        r[0] = @trf@(*((@dtype@*)&a[0 * stride]));
+        r[1] = @trf@(*((@dtype@*)&a[1 * stride]));
+        r[2] = @trf@(*((@dtype@*)&a[2 * stride]));
+        r[3] = @trf@(*((@dtype@*)&a[3 * stride]));
+        r[4] = @trf@(*((@dtype@*)&a[4 * stride]));
+        r[5] = @trf@(*((@dtype@*)&a[5 * stride]));
+        r[6] = @trf@(*((@dtype@*)&a[6 * stride]));
+        r[7] = @trf@(*((@dtype@*)&a[7 * stride]));
 
         for (i = 8; i < n - (n % 8); i += 8) {
             /* small blocksizes seems to mess with hardware prefetch */
-            NPY_PREFETCH(&a[(i + 512 / sizeof(a[0])) * stride], 0, 3);
```

Comment:
@charris, this is a good indication that the 512 used for prefetching that I picked up in #9705 was meant to measure bytes, not items